### PR TITLE
Added tests in ProfileServletTest.java and display list of pals, incomingRequests, outgoingRequests on user profile page

### PR DESCRIPTION
**Pull request by anitacu**
ProfileServletTest.java:
- Added 3 tests for sendPalRequest(), acceptPal(), and declinePal() in ProfileServlet.java

Utils.java:
- Added static getTime() function (instead of repeating method in User.java, Message.java, Conversation.java): modified AdminServlet.java to use new Utils.java

ChatServlet.java:
- Calls createConversations() if the private conversation has not been generated for two users because they were created before this feature was implemented (fixes for old users)
- If the conversation is private, only allows user to view if they are currently logged in as one of the two users in the conversation; else will print out message that they cannot view this conversation and redirect the user back to conversations/ link

ProfileServlet.java + profile.jsp:
- Displays list of pals on a user's profile page
- If user viewing his/her own page: 
-- If have notifications/pal requests from another user, added so that name links to their profile page
-- Displays list of their outgoingRequests

User.java:
- Validates user input and checks that name being added to incomingRequests and outgoingRequests is not the current user

main.css:
- Added h3 colors for profile page headers

### CODE REVIEW

File: src/main/java/codeu/model/data/Utils.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package codeu.model.data;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+
+public class Utils {
+
+    /** Returns the Instant into a String time format to display to users. */
+    public static String getTime(Instant creation) {
```
Comment (tucosh) [inline src/main/java/codeu/model/data/Utils.java]: Good! Eliminate duplicating duplications!

File: src/main/java/codeu/model/data/User.java [hunk 2]
```diff
@@ -144,8 +144,7 @@ public void setPals(List<String> pals) {
     /** Adds a pal to the list of this User's pals as long as the pal is not this User. */
     public void addPal(String name) {
         /* Validate user input & only add the pal if he/she isn't this User and is not already a pal. */
-        UserStore userStore = UserStore.getInstance();
-        if (name.matches("[\\w*\\s*]*") && userStore.isUserRegistered(name) && !name.equals(this.name) && !isPal(name)) {
+        if (name.matches("[\\w*\\s*]*") && !name.equals(this.name) && !isPal(name)) {
             this.pals.add(name);
```
Comment (tucosh) [inline src/main/java/codeu/model/data/User.java]: You realise List<String> may not be the best choice for pals. So far at least you aren't using the functionality of a List, and when you do pals.contains(..) in the isPal() method it has to iterate the entire list, hence O(n) efficiency. If add() and contains() are the only operations you need, a Set (eg HashSet) might be a better choice.

File: src/test/java/codeu/controller/ProfileServletTest.java [hunk 3]
```diff
@@ -102,9 +119,109 @@ public void testDoPostUncleanedBio() throws IOException, ServletException {
...
+        Assert.assertEquals(1, userBob.getPals().size());
+        Assert.assertEquals(1, userHelen.getPals().size());
+
+        Assert.assertEquals(0, userBob.getOutgoingRequests().size());
+        Assert.assertEquals(0, userHelen.getIncomingRequests().size());
+    }
+
+    @Test
+    public void testDeclinePal() throws IOException, ServletException {
+        User userBob = new User(
```
Comment (tucosh) [inline src/test/java/codeu/controller/ProfileServletTest.java]: Good tests .. you may want to move some of this common setup into a helper method

File: src/main/java/codeu/controller/ChatServlet.java [hunk 4]
```diff
@@ -89,12 +89,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
     Conversation conversation = conversationStore.getConversationWithTitle(conversationTitle);
     if (conversation == null) {
-      // couldn't find conversation, redirect to conversation list
-      System.out.println("Conversation was null: " + conversationTitle);
-      response.sendRedirect("/conversations");
-      return;
+      // check if this conversationTitle was a private conversation and if it doesn't exist, reason is because
+      // this was an old User before private conversations were made, so call createConversations() for both Users
+      if (conversationTitle.contains("-")) {
+        int dashPos = conversationTitle.indexOf("-");
```
[the remark this answers is not in the document]
Comment (anitacu) [reply src/main/java/codeu/controller/ChatServlet.java]: oh I see, thank you for the suggestion! I modified these lines a bit to use regex instead :)

File: src/main/java/codeu/controller/ChatServlet.java [hunk 5]
```diff
@@ -89,12 +89,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
+        String name1 = conversationTitle.substring(0, dashPos);
+        String name2 = conversationTitle.substring(dashPos + 1, conversationTitle.length());
+        User user1 = UserStore.getInstance().getUser(name1);
+        User user2 = UserStore.getInstance().getUser(name2);
+        user1.createConversations();
+        user2.createConversations();
+        conversation = conversationStore.getConversationWithTitle(conversationTitle);
```
[the remark this answers is not in the document]
Comment (anitacu) [reply src/main/java/codeu/controller/ChatServlet.java]: that is a good point, I added to check that user != null

**Pull request merged**